### PR TITLE
Remove javascript sanitization from servant-foreign (was: Fix javascript function name generation)

### PR DESCRIPTION
**Pull request by berdario**
There's a lot of duplication in JS/Vanilla, JS/Jquery, etc.etc. but fixing that would be out of scope to fix this issue

Also, you might want to tweak the logic in `toValidFunctionName`. In particular, I think it could make more sense to have it as `Text -> [Text]`, rather than `Text -> Text`, so every invalid character (`-`, `.`, weird unicodes, spaces, etc) would eventually a CamelCase or snake_case in the following character

### CODE REVIEW

**Comment by berdario** [comment]
Ok, I need to fix the tests

File: servant-foreign/test/Servant/ForeignSpec.hs [hunk 1]
```diff
@@ -19,8 +19,6 @@ camelCaseSpec = describe "camelCase" $ do
   it "converts FunctionNames to camelCase" $ do
     camelCase (FunctionName ["post", "counter", "inc"])
       `shouldBe` "postCounterInc"
-    camelCase (FunctionName ["get", "hyphen-ated", "counter"])
```
Comment (phadej) [inline servant-foreign/test/Servant/ForeignSpec.hs]: why this is removed?

Comment (phadej) [inline servant-foreign/test/Servant/ForeignSpec.hs]: And you should add a test for the case PR fixes as well

**Comment by berdario** [comment]
@phadej 
It's removed since the code it was testing does not exist anymore: that code does not belong in servant-foreign, but rather in servant-js

I want to add a few tests, but currently there's no infrastructure in place for some integration tests (which are the only reasonable ones to test that our generated code is parsable), so it'll take a bit of time to get done

my idea is to add `node` in travis, check the presence of the node binary and skip the tests otherwise, and then `node -c` to syntax check the generated code without executing it

I think it would be better/more manageable to have a few smaller PRs, rather than a huge code drop that will encompass everything

**Comment by phadej** [comment]
I cannot comment on `node` (I am not a maintainer, and IIRC someone was against complicating CI setup), but you could at least add an example which broke your code.

**Comment by berdario** [comment]
Sure, the example I used is the same one that I wrote in the [linked issue](https://github.com/haskell-servant/servant/issues/191#issuecomment-131757316)

You can write a route in servant accessible at `"foo-bar\66352baz\6158quux"`: making requests at it will work just fine, but the generated js function name will be invalid.

There might be other characters that might lead to generation of invalid code, but this PR at least is a starting point to go back from special-casing issues in a few different places in the code...

I'd also like to see what someone else thinks about changing the type of that function to `Text -> [Text]`... that would be one of the next few steps I had in mind

**Comment by jkarni** [comment]
Thanks!
I'm in favor of the changes, but like @phadej I'd like for the original issue to be put in a test, even if integration tests would eventually make more sense.

**Comment by berdario** [comment]
Note for myself:
use `Language.ECMAScript3.Parser.parse` and `Language.ECMAScript3.Lexer.identifier`... pass the generated identifier through it, and the output should be the same as the input

**Comment by phadej** [comment]
Migrated to https://github.com/haskell-servant/servant-js/pull/1

**Comment by berdario** [comment]
I realized that this should still be merged (I fixed the merge conflicts, now) since the changes to servant-foreign cannot be migrated to servant-js

(they are just removal changes, since sanitization of hyphens is something language-specific that doesn't really belong in servant-foreign)

**Comment by phadej** [comment]
@berdario True, sorry, i mixed up the pr's. Could you change the title of this one?

**Comment by phadej** [comment]
Merged in #668

**Pull request closed without merging**